.circleci/config.yml

```diff
@@ -0,0 +1,75 @@
+version: 2.1
+
+executors:
+  default:
+    docker:
+      - image: circleci/golang:1.12
+
+jobs:
+  build:
+    executor:
+      name: default
+    steps:
+      - checkout
+      - run:
+          name: Setup Environment Variables
+          command: |
+            echo "export PATH=$GOPATH/bin:$PATH" >> $BASH_ENV
+            echo "export PATH=$PATH:$PWD/bin" >> $BASH_ENV
+            echo "HUGO_VERSION='0.58.3'" >> $BASH_ENV
+            echo "MATTERMOST_SERVER_RELEASE_TAG=v5.17.1" >> $BASH_ENV
+            source $BASH_ENV
+      - run:
+          name: install Node JS
+          command: |
+            sudo apt-get update
+            sudo apt-get install -y nodejs npm
+      - run:
```

Comment:
checking the travis.yaml config i dont see all these steps there. why we need those?

Reply:
### What Was Failing

Current `mattermost-server` master has breaking changes which cause script to fail.

https://travis-ci.org/mattermost/mattermost-developer-documentation/builds/620266705#L617

Based on the Travis CI output above the `scripts/plugin-godocs.go` fails after the
`mattermost-server` is installed in Makefile `backend-plugin-data`.
https://github.com/mattermost/mattermost-developer-documentation/blob/master/Makefile#L10

The breaking change to `mattermost-server` was introduced 5 days ago in the https://github.com/mattermost/mattermost-server/commit/2259b7f2a8b92b5dada097e77c284d5d14a01ba3

As the `mattermost-server` is not explicitly required in the `imports` of `scripts/plugin-godocs.go` , but during runtime this causes missing paths and failure of script. The example is in the Travis output.

I've tried to fix this issue using go mod `replace` directive to just point to  proper working release. So that we could use current `Makefile` with `make dist` like in the `travis.yml`. But this approach fails.
You can validate below.
The pipeline code would look like this:
https://github.com/mattermost/mattermost-developer-documentation/compare/master...bpietraga:test-2?expand=1
You can look failing pipeline for this `test-2` branch here:
https://circleci.com/gh/bpietraga/mattermost-developer-documentation/151

### Fix

I've solved this, using git checkout and rolling to release tag here:
https://github.com/mattermost/mattermost-developer-documentation/pull/450/files#diff-1d37e48f9ceff6d8030570cd36286a61R34

And then adding the steps from `Makefile`

I know that this is not ideal solution as the current `Makefile` commands fail.

Reply:
should we fix this in the source and not doing workarounds?

Reply:
will be fixed here: https://github.com/mattermost/mattermost-developer-documentation/pull/452

Reply:
It is hard to not to agree with this one 😄 
Great! It is merged, so I've rebased the code with clean code. 
This time no workarounds. New commit: https://github.com/mattermost/mattermost-developer-documentation/pull/450/commits/e5870b7142a88d8cd7e4b91610efd67113f9f33f

Reply:
your solution was just an workaround and that will not work for the long run.